### ISSUE TITLE
org-msg is not activated with mu4e-compose-reply

### ISSUE THREAD

**Issue by baracunatana** [jeremy-compostella/org-msg#153]
Hi everyone.

I have the same issue as https://github.com/jeremy-compostella/org-msg/issues/87 after updating my pc (it was working fine two days ago<sup>TM</sup>). Composing a new email (with `mu4e-compose-new`) works as intended, but `mu4e-compose-reply` doesn't activate `org-msg` (however, `org-msg-mode` evaluates to `t` in the reply buffer). I can call org-msg-edit-mode and it kind of works (I can use org to edit my message), but it doesn't add my signature and the original message is displayed whit > and not with the outlook style.

I tried going to the previous commit or org-msg (https://github.com/jeremy-compostella/org-msg/commit/77f5911b7d390a069104db20be86293506ffbff2) but it didn't work.

I have `mu4e` 1.8.1, `org-msg` 4.0 and `emacs` 29.05. I also set `mail-user-agent` to `'mu4e-user-agent` in my config file.

Any help is appreciated.

**Comment by Phundrak** [comment]
I have the same issue, but with some differences:
- If I load mu4e before org-mode, org-msg isn’t loaded at all and `org-msg-mode` evaluates to `nil` in the reply buffer
- If I load org-mode, then org-msg is loaded correctly and works as expected
  
For the record, here are the relevant software versions:
- mu & mu4e 1.8.1 (installed from the AUR on Arch Linux)
- org-msg 4.0 (commit 60e22e4, master’s HEAD at the time of this message)
- Emacs 29.0.50 (commit https://github.com/emacs-mirror/emacs/commit/1dc70544d8)

Here is my relevant `org-msg` configuration (stripped of a few lines, [full version available here](https://labs.phundrak.com/phundrak/dotfiles/src/branch/master/org/config/emacs.org#Packages-Configuration-Applications-Email-Composing-messagesth71fl6184j0)):
``` emacs-lisp
(use-package org-msg
  :after (org mu4e)
  :straight (:build t)
  :hook ((mu4e-compose-pre . org-msg-mode))
  :config
  (setq org-msg-default-alternatives '((new           . (text html))
                                       (reply-to-html . (text html))
                                       (reply-to-text . (text)))))
```

**Comment by baracunatana** [comment]
@Phundrak Thanks for the help. I tried your customization for `org-msg-default-alternatives` and `org-msg` is now launched when replying to a message. However, it isn't converting the citation to outlook format (i.e. I still have the > citation style). Also, the signature is added after the citation. I have `org-msg-convert-citation` set to `t`, but still no luck. Do you have the same issue?

**Comment by Phundrak** [comment]
> @Phundrak Thanks for the help. I tried your customization for
> `org-msg-default-alternatives` and `org-msg` is now launched when
> replying to a message.

I still have the issue of org-msg not firing up if org-mode is not
loaded on my side.

> However, it isn't converting the citation to outlook format (i.e. I
> still have the > citation style). Also, the signature is added after
> the citation. I have `org-msg-convert-citation` set to `t`, but
> still no luck. Do you have the same issue?

I completely forgot I had org-msg-convert-citation set to t too, turns
out it never worked, I always had the greater-than citation style. But
I guess this is a separated issue from the current one.

--
Lucien “Phundrak” Cartier-Tilet
<https://phundrak.com> (Français)
<https://phundrak.com/en> (English)
Sent from GNU/Emacs

**Comment by baracunatana** [comment]
I can confirm that going back to `mu4e` `1.6.10` (from `1.8.1`) solves the issue. It seems that recent changes in `mu4e` interfered with `org-msg` in some way.

**Comment by baracunatana** [comment]
> I still have the issue of org-msg not firing up if org-mode is not loaded on my side.

What if you configure it with only  `:after (mu4e)`

**Comment by benthamite** [comment]
I also started experiencing this issue after upgrading to the latest version of `mu4e` (1.8.5). @baracunatana, where you able to identify the cause of the problem, or are you still using an earlier version of `mu4e`?

**Comment by baracunatana** [comment]
@benthamite My debugging skill are not up to this challenge, so I just went back to `mu` 1.6.10 and it works perfectlly now.

**Comment by ifinkelstein** [comment]
Also having the same issue with ~mu4e~ 1.8.5. Will watch this issue to see if there's a resolution.

**Comment by benthamite** [comment]
I confirm that downgrading to 1.6.11 solved the issue for me. FYI, [here's the mu 1.6.11 homebrew formula](https://raw.githubusercontent.com/Homebrew/homebrew-core/c42744ba370044e5363e1a6547014c810304b520/Formula/mu.rb). To downgrade with brew, save the file, then
1. `$ brew uninstall mu`
2. `$ brew install /path/to/mu.rb`
3. `$ brew pin mu`

**Comment by bergheim** [comment]
@Phundrak is this still working for you? I tried to apply your config, but I still get the ">" quotation. org syntax works, however.

**Comment by tiZero** [comment]
My guess: under mu 1.8.* I cannot retrieve the fields `:body-txt` or `:body-html` from message property list anymore. Expressions like `(mu4e-message-field msg :body-html)` (as in `org-msg-article-htmlp-mu4e`) evaluate to `nil` and org-msg selects the `reply-to-text` alternative in every case.

**Comment by cnoceda** [comment]
> My guess: under mu 1.8.* I cannot retrieve the fields `:body-txt` or `:body-html` from message property list anymore. Expressions like `(mu4e-message-field msg :body-html)` (as in `org-msg-article-htmlp-mu4e`) evaluate to `nil` and org-msg selects the `reply-to-text` alternative in every case.

 I've the same issue. I check `mu4e-message-field` and it's return nil as you say.

**Comment by david-alvarez-rosa** [comment]
Could https://github.com/djcb/mu/issues/2337 be related?

**Comment by Phundrak** [comment]
Org-msg works for me, and I removed `org-msg-convert-citation' my config
so I cannot answer regarding this issue.

--
Lucien “Phundrak” Cartier-Tilet
<https://phundrak.com> (Français)
<https://phundrak.com/en> (English)
Sent from GNU/Emacs

**Comment by bergheim** [comment]
I don't think so @david-alvarez-rosa  - this has always been like this for me.

I still cannot get org-msg to work the way it did. Replies will be text only.

Also, as @Phundrak mentions as well, `org-msg-convert-citation` has never worked for me either.

**Comment by gajama** [comment]
I've put in a PR that I think fixes the mu4e issues: https://github.com/jeremy-compostella/org-msg/pull/159

**Comment by seblemaguer** [comment]
From my understanding, #159 solves the issue. I close this issue. If this problems come back, don't hesitate to reopen it.
